open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidationUtils.java

```diff
@@ -0,0 +1,134 @@
+     *
+     * @return True if idempotency is required, else False.
+     */
+    static boolean isIdempotencyEnabledFromConfig() {
+
+        String isIdempotencyEnabled = (String) configs.get(IdempotencyConstants.IDEMPOTENCY_IS_ENABLED);
+        return Boolean.parseBoolean(isIdempotencyEnabled);
+    }
+
+    /**
+     * Method to retrieve the consent ids that have the idempotency key name and value as attribute.
+     *
+     * @param idempotencyKeyName     Idempotency Key Name
+     * @param idempotencyKeyValue    Idempotency Key Value
+     * @return   List of consent ids
+     */
+    static ArrayList<String> getConsentIdsFromIdempotencyKey(String idempotencyKeyName,
+                                                             String idempotencyKeyValue) {
+        try {
+            return consentCoreService.getConsentIdByConsentAttributeNameAndValue(
+                    idempotencyKeyName, idempotencyKeyValue);
+        } catch (ConsentManagementException e) {
+            log.debug("No consent ids found for the idempotency key value");
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Method to compare the client ID sent in the request and client id retrieved from the database.
+     *
+     * @param requestClientID     Client ID sent in the request
+     * @param dbClientId          client ID retrieved from the database
+     * @return   Whether JSON client Ids are equal
+     */
+    static boolean isClientIdsMatching(String requestClientID, String dbClientId) {
+        return requestClientID.equals(dbClientId);
+    }
+
+    /**
+     * Method to check whether difference between two dates is less than the configured time.
+     *
+     * @param createdTime Created Time of the request
+     * @return  Whether the request is received within allowed time
+     */
+    static boolean isRequestReceivedWithinAllowedTime(long createdTime) {
+
+        if (createdTime == 0L) {
+            return false;
+        }
+        String allowedTimeDuration = (String) configs.get(IdempotencyConstants.IDEMPOTENCY_ALLOWED_TIME);
+        if (allowedTimeDuration != null) {
+            OffsetDateTime createdDate = OffsetDateTime.parse(convertToISO8601DateTimeFormat(createdTime));
+            OffsetDateTime currDate = OffsetDateTime.now(createdDate.getOffset());
+
+            long diffInHours = Duration.between(createdDate, currDate).toMinutes();
+            return diffInHours <= Long.parseLong(allowedTimeDuration);
+        } else {
+            log.error("Idempotency Allowed duration is null");
+            return false;
+        }
+    }
+
+    /**
+     * Convert long date values to ISO 8601 format.
+     * @param dateValue     Date value
+     * @return ISO 8601 formatted date
+     */
+    private static String convertToISO8601DateTimeFormat(long dateValue) {
+
+        DateFormat simple = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX");
+        Date simpleDateVal = new Date(dateValue * 1000);
```

Comment:
Could dateValue be 0?

Reply:
it is already check before calling this method